src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,132 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int DEFAULT_CAPACITY = 16;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size > checkThreshold()) {
+            resize();
+        }
```

Comment:
method  with name checkSomething should return boolean or void, we can transfer if{} logic inside this method
otherwise it should be getThreshold